command/search.go

```diff
@@ -0,0 +1,103 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/HotelsDotCom/flyte-client/flyte"
+	"github.com/HotelsDotCom/flyte-jira/client"
+	"github.com/HotelsDotCom/flyte-jira/domain"
+	"log"
+)
+
+var (
+	searchSuccessEventDef = flyte.EventDef{Name: "SearchSuccess"}
+	searchFailureEventDef = flyte.EventDef{Name: "SearchFailure"}
+)
+
+var SearchIssuesCommand = flyte.Command{
+	Name:         "SearchIssues",
+	OutputEvents: []flyte.EventDef{searchSuccessEventDef, searchFailureEventDef},
+	Handler:      searchIssuesHandler,
+}
+
+func searchIssuesHandler(rawInput json.RawMessage) flyte.Event {
+
+	input := SearchIssuesInput{"", 0, 10}
+	if err := json.Unmarshal(rawInput, &input); err != nil {
+		err := fmt.Errorf("input is not valid: %s", err)
+		return flyte.NewFatalEvent(err)
+	}
+
+	if input.Query == "" {
+		err := fmt.Errorf("Empty query string")
+		return newSearchFailureEvent(input.Query, input.StartIndex, input.MaxResults, err.Error())
+	}
+
+	searchResult, err := client.SearchIssues(input.Query, input.StartIndex, input.MaxResults)
+	if err != nil {
+		err := fmt.Errorf("Could not search for issues: %v", err)
+		log.Println(err)
+		return newSearchFailureEvent(input.Query, input.StartIndex, input.MaxResults, err.Error())
+	}
+
+	return newSearchSuccessEvent(
+		input.Query,
+		input.StartIndex,
+		input.MaxResults,
+		searchResult.TotalResults,
+		searchResult.Issues)
+}
+
+func newSearchSuccessEvent(query string, startIndex int, maxResults int, totalResults int, unformattedIssues []domain.Issue) flyte.Event {
+
+	inputDetails := SearchIssuesInput{query, startIndex, maxResults}
+	var issues []IssuePayload
+	for _, issue := range unformattedIssues {
+		formattedIssue := IssuePayload{
+			Id:          issue.Key,
+			Summary:     issue.Fields.Summary,
+			Status:      issue.Fields.Status.Name,
+			Description: issue.Fields.Description,
+			Assignee:    issue.Fields.Assignee.Name,
+		}
+		issues = append(issues, formattedIssue)
+	}
+	return flyte.Event{
+		EventDef: searchSuccessEventDef,
+		Payload:  SearchSuccessOutput{inputDetails, totalResults, issues},
+	}
+}
+
+func newSearchFailureEvent(query string, startIndex int, maxResults int, error string) flyte.Event {
```

Comment:
Can we use error as error instead of a string? I think it makes more sense.
Would it make sense to pass SearchIssuesInput instead of instantiate it again?

Reply:
Switched it to an error type and changed the parameters to use the SearchIssueInput type. 
I believe it looks less cluttered this way.